wfc_dylib/src/lib.rs

```diff
@@ -417,6 +435,7 @@ pub extern "C" fn wfc_rng_state_free(wfc_rng_state_handle: WfcRngStateHandle) {
 pub enum WfcObserveResult {
     Deterministic = 0,
     Contradiction = 1,
+    Nondeterministic = 2,
```

Comment:
Now that this is reused across multiple functions, maybe we should call it something else? Maybe `WfcWorldStatus`? Also see the main review comment - if we to remove `wfc_world_status`, this can stay the way it is.